org.eclipse.xtext.util/src/org/eclipse/xtext/util/UriExtensions.java

```diff
@@ -34,7 +34,7 @@ public URI toUri(String stringUri) {
 	 */
 	public URI toEmfUri(java.net.URI netUri) {
 		String decoded = toDecodedString(netUri);
-		URI uri = URI.createURI(decoded, false);
+		URI uri = URI.createURI(decoded, true);
```

Comment:
Can you help me understand the use case?
We receive a valid, encoded java.net.URI.
We decode it to something (which might now contain a % sign). Whether it contains it or not, is not relevant though - the string is supposed to be _decoded_.
Now we make an EMF URI from it: The decoded string must in any case be encoded. So `ignoreEscaped=true` appears to be wrong to me.

Reply:
@szarnekow: The use case is that you create a file on the client with something that contains as part of the name something that looks like an scaped character, for example "WyEx%23mpleFile.whatever".
When the client encodes it, it will transform it to "WyEx%2523mpleFile.whatever".
Unless you apply my fix, the uri on the server side, ends up being "WyEx#mpleFile.whatever" and not the original one, because we decode twice and not once (as I think we should).

I discover this problem because we have files with '#' in the name, and the decoding would think #... was the fragment part of an EMF URI. One of my first tries was to replace '#' with '%23' in the files names, and was surprised to still get '#' as a fragment. Looking at the code, it was due to the double decoding.

After looking at it later in more detail, I have discovered there is a another bug,  because in LSP '#' is not a fragment separator, just a normal character, so as part of this PR I have fixed both problems and created two different tests for them.

Reply:
Can you by chance add an integration test with a filename that contains a `#` char`

Reply:
I have added one.